include/drivers/gpio.h

```diff
@@ -30,14 +30,59 @@ extern "C" {
  * @brief GPIO Driver APIs
  * @defgroup gpio_interface GPIO Driver APIs
  * @ingroup io_interfaces
+ *
+ * @{
+ * @name GPIO Atomicity and Interrupt-Related Behavior
+ *
  * @{
+ * All GPIO peripheral API operations satisfy the following promises:
+ *
+ * * All functions shall be thread-safe and interrupt-safe.
+ * * All functions may be invoked from interrupt service routines and
+ *   callbacks.  Functions that cannot fulfill their behavioral
```

Comment:
We repeat ourselves. We've mentioned already that the functions are interrupt safe and then we say "All functions may be invoked from interrupt service routines and callbacks". I would remove the sentence.

Reply:
No; "interrupt safe" and "can be invoked from ISRs and callbacks" are two different things.